src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -505,7 +505,8 @@ public async Task CreateGoodBreakpointAndHitGoToNonWasmPageComeBackAndHitAgain(s
                 expression = "window.setTimeout(function() { " + func_name + "(); }, 1);"
             });
             await cli.SendCommand("Runtime.evaluate", run_method, token);
-            await Task.Delay(1000, token);
+            insp.NotifyOfNonWasmIsLoading(true);
+            await insp.WaitFor(Inspector.NON_WASM_PAGE_READY);
```

Comment:
You can use `insp.WaitForEvent(..)` for this. We do this for waiting for breakpoint resolved events too. I would suggest:

```diff
diff --git a/src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs b/src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs
index bb228900f4a..69a99a3fb72 100644
--- a/src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs
+++ b/src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs
@@ -505,8 +505,7 @@ public async Task CreateGoodBreakpointAndHitGoToNonWasmPageComeBackAndHitAgain(s
                 expression = "window.setTimeout(function() { " + func_name + "(); }, 1);"
             });
             await cli.SendCommand("Runtime.evaluate", run_method, token);
-            insp.NotifyOfNonWasmIsLoading(true);
-            await insp.WaitFor(Inspector.NON_WASM_PAGE_READY);
+            await WaitForEventAsync("Runtime.executionContextCreated");

             run_method = JObject.FromObject(new
             {
diff --git a/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs b/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs
index b533d39910a..6d40816626c 100644
--- a/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs
+++ b/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs
@@ -1451,17 +1451,17 @@ internal async Task<JObject> LoadAssemblyAndTestHotReload(string asm_file, strin
             return await WaitFor(Inspector.PAUSE);
         }

-        public async Task<JObject> WaitForBreakpointResolvedEvent()
+        public Task<JObject> WaitForBreakpointResolvedEvent() => WaitForEventAsync("Debugger.breakpointResolved");
+
+        public async Task<JObject> WaitForEventAsync(string eventName)
         {
             try
             {
-                var res = await insp.WaitForEvent("Debugger.breakpointResolved");
-                _testOutput.WriteLine ($"breakpoint resolved to {res}");
-                return res;
+                return await insp.WaitForEvent(eventName);
             }
             catch (TaskCanceledException)
             {
-                throw new XunitException($"Timed out waiting for Debugger.breakpointResolved event");
+                throw new XunitException($"Timed out waiting for {eventName} event");
             }
         }
```